orttraining/orttraining/python/training/orttrainer.py

```diff
@@ -787,7 +787,12 @@ def _training_session_run_helper(self, is_train, inputs, inputs_desc, outputs_de
         outputs_desc_resolved = self._resolve_symbolic_dimensions(inputs, inputs_desc, outputs_desc)
         result = {}
         for output_desc in outputs_desc_resolved:
-            torch_tensor = torch.zeros(output_desc.shape, device=self.options.device.id,
+            target_device = self.options.device.id
+            if self.options.mixed_precision.enabled and output_desc.name == self.model_desc.all_finite.name:
```

Comment:
>self.options.mixed_precision.enabled [](start = 15, length = 36)

Why need to check if mixed_precision is enabled? Seems IsAllFinite covers fp32 as well.

Reply:
It seems like `self.model_desc.all_finite.name` is not defined when mixed precision is disabled. How is all_finite used in fp32 case if loss scaling is only applied for mixed precision calculations? Is NaN gradients a particular problem for fp16 cases but not fp32?

Reply:
Current design allows all_finite for mixed_precision only.